### ISSUE TITLE
IDISP006 gets incorrectly triggered in IAsyncDisposable

### ISSUE THREAD

**Issue by jeremyVignelles** [DotNetAnalyzers/IDisposableAnalyzers#222]
Quick issue, didn't take time to check if it is reproducible or something, but got IDISP006 on _senderQueueSemaphore :

```cs
    public class MessageSender<TMessage> : IAsyncDisposable
    {
        private readonly SemaphoreSlim _sendQueueSemaphore = new SemaphoreSlim(1);

        private List<TMessage> _sendQueue = new List<TMessage>();

        public async ValueTask DisposeAsync()
        {
            await Task.CompletedTask.ConfigureAwait(false);
        }
    }
```

I'm just building my app, so obviously I did something wrong, but IDisposable is implemented, it just don't Dispose the value? Or am I missing the difference with IDISP002 (which is also triggered) ?

**Comment by JohanLarsson** [comment]
Thanks for reporting, adding a test.

**Comment by JohanLarsson** [comment]
The warning goes away when the member is handled in `DisposeAsync`

No warning for the following code:

```cs
namespace N
{
    using System;
    using System.IO;
    using System.Threading.Tasks;

    public class C : IAsyncDisposable
    {
        private readonly IAsyncDisposable disposable = File.OpenRead(string.Empty);

        public async ValueTask DisposeAsync()
        {
            await this.disposable.DisposeAsync();
        }
    }
}
```

Should we change it so that the warning goes away when we implement `IAsyncDisposable`?
I'm thinking current behavior is perhaps useful as it keeps the fix that generates the `Disposable` b loat available, not sure.
